core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1129,6 +1129,15 @@ private[spark] object Utils extends Logging {
     s"${TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTimeNs)} ms"
   }
 
+  /**
+   * Lists files recursively.
+   */
+  def recursiveList(f: File): Array[File] = {
+    require(f.isDirectory)
+    val current = f.listFiles
+    current ++ current.filter(_.isDirectory).flatMap(recursiveList)
```

Comment:
nit: I think this can be done easily using non-recursive approach?

Reply:
Sure, done in f5a6104